### ISSUE TITLE
Tables with mismatched headers don't render (but did in ancient marked 0.7.0?)

### ISSUE THREAD

**Issue by gardnerjr** [markedjs/marked#2745]
**Marked version:**
4.x
Ancient code on marked 0.7.0 seems to handle this better

**Describe the bug**
A  table with mismatched headers and rows won't render:

```
| One header |
|:------------- |:-------------|:-----|
| row | has | 3 cols |

```
We have a site using an ancient verison of marked, ~0.7.0, where this works in a strange way, which generates this:

![image](https://user-images.githubusercontent.com/10158007/222019949-a64a21f4-1a01-4c7b-bf2a-5aca5ea676e2.png)

```
<table>
<thead>
<tr>
<th style="text-align: left;">One header</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">row</td>
<td style="text-align: left;">has</td>
<td style="text-align: left;">3 cols</td>
</tr>
</tbody>
</table>
```

**To Reproduce**

1. [Marked broken example](https://marked.js.org/demo/?text=%7C%20One%20header%20%7C%0A%7C%3A-------------%20%7C%3A-------------%7C%3A-----%7C%0A%7C%20row%20%7C%20has%20%7C%203%20cols%20%7C%0A&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)
    if you add a couple `|` to the first line to have matching number of columns, the table will start working
2. CommonMark, DaringFireball don't support tables

**Expected behavior**
The ancient behavior of creating a table with a partial header was much more forgiving.  In this case, the second header line specifying the alignment is correct and i'd expect that to be the thing that "wins".

**Comment by gardnerjr** [comment]
Actually i tried with the other versions on the demo site, it looks like i have to go all the way back to  [0.3.19](https://marked.js.org/demo/?text=%7C%20One%20header%20%7C%0A%7C%3A-------------%20%7C%3A-------------%7C%3A-----%7C%0A%7C%20row%20%7C%20has%20%7C%203%20cols%20%7C%0A&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=0.3.19) to get an example of it working like would be most forgiving

**Comment by calculuschild** [comment]
Yep, this is a result of Marked following the spec more closely in later versions. Ancient Marked mistakenly allowed bad table syntax. Newer Marked is better at following the rules.
